src/v/kafka/client/cluster.cc

```diff
@@ -222,22 +369,27 @@ ss::future<> cluster::apply_metadata(metadata_response reply) {
       _logger.debug,
       "Applying metadata response with {{ cluster_id: {}, "
       "controller_id: {}, brokers: {} and {} topics }}",
-      reply.data.cluster_id,
-      reply.data.controller_id,
-      reply.data.brokers,
-      reply.data.topics.size());
+      reply.cluster_id,
+      reply.controller_id,
+      reply.brokers,
+      reply.topics.size());
 
-    _cluster_id = reply.data.cluster_id;
-    if (reply.data.controller_id == unknown_node_id) {
+    _cluster_id = reply.cluster_id;
+    _cluster_authorized_operations = reply.cluster_authorized_operations;
+    if (reply.controller_id == unknown_node_id) {
         _controller_id.reset();
     } else {
-        _controller_id = reply.data.controller_id;
+        _controller_id = reply.controller_id;
+    }
+    if (!reply.topics.empty()) {
```

Comment:
this logic seems to be incorrect, what if all topics in the cluster were simply removed ?